e2e/interchaintestv8/e2esuite/utils.go

```diff
@@ -205,3 +247,49 @@ func IsLowercase(s string) bool {
 	}
 	return true
 }
+
+func (s *TestSuite) GetUnionClientState(ctx context.Context, clientID string) (*ibcwasmtypes.ClientState, ethereumligthclient.ClientState) {
+	simd := s.ChainB
+	clientStateResp, err := GRPCQuery[clienttypes.QueryClientStateResponse](ctx, simd, &clienttypes.QueryClientStateRequest{
+		ClientId: clientID,
+	})
+	s.Require().NoError(err)
+
+	var clientState ibcexported.ClientState
+	err = simd.Config().EncodingConfig.InterfaceRegistry.UnpackAny(clientStateResp.ClientState, &clientState)
+	s.Require().NoError(err)
+
+	wasmClientState, ok := clientState.(*ibcwasmtypes.ClientState)
+	s.Require().True(ok)
+	s.Require().NotEmpty(wasmClientState.Data)
+
+	var ethClientState ethereumligthclient.ClientState
+	err = simd.Config().EncodingConfig.Codec.Unmarshal(wasmClientState.Data, &ethClientState)
+	s.Require().NoError(err)
+
+	return wasmClientState, ethClientState
+}
+
+func (s *TestSuite) GetUnionConsensusState(ctx context.Context, clientID string, height clienttypes.Height) (*ibcwasmtypes.ConsensusState, ethereumligthclient.ConsensusState) {
+	simd := s.ChainB
+	consensusStateResp, err := GRPCQuery[clienttypes.QueryConsensusStateResponse](ctx, simd, &clienttypes.QueryConsensusStateRequest{
+		ClientId:       clientID,
+		RevisionNumber: height.RevisionNumber,
+		RevisionHeight: height.RevisionHeight,
+		LatestHeight:   false,
+	})
+	s.Require().NoError(err)
+
+	var consensusState ibcexported.ConsensusState
+	err = simd.Config().EncodingConfig.InterfaceRegistry.UnpackAny(consensusStateResp.ConsensusState, &consensusState)
+	s.Require().NoError(err)
+
+	wasmConsenusState, ok := consensusState.(*ibcwasmtypes.ConsensusState)
+	s.Require().True(ok)
+
+	var ethConsensusState ethereumligthclient.ConsensusState
+	err = simd.Config().EncodingConfig.Codec.Unmarshal(wasmConsenusState.Data, &ethConsensusState)
+	s.Require().NoError(err)
+
+	return wasmConsenusState, ethConsensusState
+}
```

Comment:
Maybe there is a better place for these functions? I'd also like them to take the cosmos chain as an argument instead of assuming it is ChainB

Reply:
They could go into light_clients.go, I suppose. But I feel like they are just as natural to have here. But no strong feelings on that. Changing to accept cosmos chain argument.